railties/lib/rails/commands/stats/stats_command.rb

```diff
@@ -1,41 +1,14 @@
 # frozen_string_literal: true
 
-# While global constants are bad, many 3rd party tools depend on this one (e.g
-# rspec-rails & cucumber-rails). So a deprecation warning is needed if we want
-# to remove it.
-STATS_DIRECTORIES ||= [
-  %w(Controllers        app/controllers),
-  %w(Helpers            app/helpers),
-  %w(Jobs               app/jobs),
-  %w(Models             app/models),
-  %w(Mailers            app/mailers),
-  %w(Mailboxes          app/mailboxes),
-  %w(Channels           app/channels),
-  %w(Views              app/views),
-  %w(JavaScripts        app/assets/javascripts),
-  %w(Stylesheets        app/assets/stylesheets),
-  %w(JavaScript         app/javascript),
-  %w(Libraries          lib/),
-  %w(APIs               app/apis),
-  %w(Controller\ tests  test/controllers),
-  %w(Helper\ tests      test/helpers),
-  %w(Job\ tests         test/jobs),
-  %w(Model\ tests       test/models),
-  %w(Mailer\ tests      test/mailers),
-  %w(Mailbox\ tests     test/mailboxes),
-  %w(Channel\ tests     test/channels),
-  %w(Integration\ tests test/integration),
-  %w(System\ tests      test/system),
-]
-
 module Rails
   module Command
     class StatsCommand < Base # :nodoc:
       desc "stats", "Report code statistics (KLOCs, etc) from the application or engine"
       def perform
         require "rails/code_statistics"
+        boot_application!
```

Comment:
Do we need to boot the app still?

Reply:
In the tests I call `add_directory` in a `config/initializer` which requires the app to boot I think.
In a real world app `add_directory` probably gets called by a gem which requires Bundler to initialize?

For Rake tasks we load all tasks from dependencies so we can extend the Rails tasks:
https://github.com/rails/rails/blob/9c8390032c85df23d973432b67d025df05c838f7/railties/lib/rails/engine.rb#L687-L690

I don't think we support anything like that for Thor.

Reply:
I see. Lets keep the boot otherwise we can't config it in the application.